ranger-schema-registry-plugin/authorizer-shim/src/main/java/com/hortonworks/registries/ranger/authorization/schemaregistry/authorizer/RangerSchemaRegistryAuthorizer.java

```diff
@@ -0,0 +1,240 @@
+package com.hortonworks.registries.ranger.authorization.schemaregistry.authorizer;
+
+import org.apache.ranger.plugin.classloader.RangerPluginClassLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import java.lang.reflect.Constructor;
+import java.util.Set;
+
+public class RangerSchemaRegistryAuthorizer implements Authorizer {
```

Comment:
I prefer the hive shim implementation, where you have a factory class and it returns an instance of Authorizer. Let me know your thoughts about it.

Reply:
I think that we need to keep the current implementation. The classloader should be activated not only during authorizer instantiation. It should be also activated when the other methods are called (authorizeSchemaGroup, authorizeSchema, authorizeSchemaBranch and etc.). Probably this fact was not taken into account when the hive plugin was created. The other plugins (kafka, atlas, hbase, hdfs, kms and etc.) use the current (non-hive) implementation.

Reply:
The factory method is added that allows using of custom authorizer.